scripts/bumpfee.py

```diff
@@ -0,0 +1,254 @@
+                    )
+                )
+            )[0]
+        )
+    if len(_mixdepths) != 1:
+        raise ValueError('Transaction inputs must be from the same mixdepth.')
+
+def tx_vsize(tx):
+    raw_tx_size = len(tx.serialize())
+    witness_size = len(tx.wit.serialize())
+    non_witness_size = raw_tx_size - witness_size
+    return int(non_witness_size + .25 * witness_size)
+
+def compute_bump_fee(tx, tx_info, fee_rate):
+    tx_size_n = tx_vsize(tx)
+    tx_fee = btc.amount_to_sat(abs(tx_info['fee']))
+    proposed_fee_rate = fee_rate / Decimal(1000.0)
+    proposed_fee = int(tx_size_n * proposed_fee_rate)
+    min_proposed_fee = tx_fee + tx_size_n
+    min_proposed_fee_rate = min_proposed_fee / Decimal(tx_size_n)
+
+    if proposed_fee < (tx_fee + tx_size_n):
+        raise ValueError('Proposed fee for transaction replacement: '
+                         '%d sats (%.1f sat/vB) is below minimum required '
+                         'for relay: %d sats (%.1f sat/vB). '
+                         'Try using a higher fee setting.'
+                         % (proposed_fee, proposed_fee_rate, 
+                            min_proposed_fee, min_proposed_fee_rate))
+
+    return (proposed_fee - tx_fee)
+
+def prepare_transaction(new_tx, old_tx, wallet):
+    input_scripts = {}
+    spent_outs = []
+    for ix, vin in enumerate(new_tx.vin):
+        script = wallet.pubkey_to_script(
+            btc.extract_pubkey_from_witness(old_tx, ix)[0])
+        tx_info = jm_single().bc_interface.get_transaction(
+            hextobin(btc.b2lx(new_tx.vin[ix].prevout.hash)))
+        prev_tx = btc.CTransaction.deserialize(hextobin(tx_info['hex']))
+        amount = prev_tx.vout[new_tx.vin[ix].prevout.n].nValue
+
+        input_scripts[ix] = (script, amount)
+        spent_outs.append(btc.CMutableTxOut(amount, script))
+
+    return (input_scripts, spent_outs)
+
+def sign_transaction(new_tx, old_tx, wallet_service):
+    input_scripts, _ = prepare_transaction(new_tx, old_tx, wallet_service.wallet)
+    success, msg = wallet_service.sign_tx(new_tx, input_scripts)
+    if not success:
+        raise RuntimeError("Failed to sign transaction, quitting. Error msg: " + msg)
+
+def sign_psbt(new_tx, old_tx, wallet_service):
+    _, spent_outs = prepare_transaction(new_tx, old_tx, wallet_service.wallet)
+    unsigned_psbt = wallet_service.create_psbt_from_tx(
+        new_tx, spent_outs=spent_outs)
+    print(unsigned_psbt.to_base64())
+    signed_psbt, err = wallet_service.sign_psbt(unsigned_psbt.serialize())
+
+    if err:
+        raise RuntimeError("Failed to sign PSBT, quitting. Error message: " + err)
+
+    return btc.PartiallySignedTransaction.deserialize(signed_psbt)
+
+def create_bumped_tx(tx, wallet, fee):
+    for ix, vout in enumerate(tx.vout):
+        if wallet.is_known_script(vout.scriptPubKey):
+            # check if the output is a change address
+            if wallet.script_to_path(vout.scriptPubKey)[-2] != 1:
```

Comment:
Since this will not return true for any timelocked output created by `FidelityBondMixin` it should be fine. Albeit I'm not sure checking `!=1` is as general as we'd like. You should replace `1` with `BaseWallet.ADDRESS_TYPE_INTERNAL`.

Even then I'm not 100% sure this is a fully satisfactory check ... I'd tend to think it is better to have a way for the user to specify which output index they want to decrement.

Reply:
The vast majority of transactions that this is likely to be used for would be transactions with a change output to the same wallet and should be straightforward to select which output to deduct fees from. I could add the option to explicitly choose which output to decrement and also remove the check.